velox/functions/prestosql/types/IPPrefixType.h

```diff
@@ -15,13 +15,145 @@
+
+namespace {
+auto splitIpSlashCidr(folly::StringPiece ipSlashCidr) {
+  folly::small_vector<folly::StringPiece, 2> vec;
+  folly::split('/', ipSlashCidr, vec);
+  return vec;
+}
+
+Status handleFailedToCreateNetworkError(
+    const std::string& ipaddress,
+    folly::CIDRNetworkError error) {
+  if (threadSkipErrorDetails()) {
+    return Status::UserError();
+  }
+
+  switch (error) {
+    case folly::CIDRNetworkError::INVALID_DEFAULT_CIDR: {
+      return Status::UserError(
+          "defaultCidr must be <= std::numeric_limits<uint8_t>::max()");
+    }
+    case folly::CIDRNetworkError::INVALID_IP_SLASH_CIDR: {
+      return Status::UserError(
+          "Invalid IP address string received. Received string:{} of length:{}",
+          ipaddress,
+          ipaddress.size());
+    }
+    case folly::CIDRNetworkError::INVALID_IP: {
+      const auto vec = splitIpSlashCidr(ipaddress);
+      return Status::UserError(
+          "Invalid IP address '{}'", vec.size() > 0 ? vec.at(0) : "");
+    }
+    case folly::CIDRNetworkError::INVALID_CIDR: {
+      auto const vec = splitIpSlashCidr(ipaddress);
+      return Status::UserError(
+          "Mask value '{}' not a valid mask", vec.size() > 1 ? vec.at(1) : "");
+    }
+    case folly::CIDRNetworkError::CIDR_MISMATCH: {
+      const auto vec = splitIpSlashCidr(ipaddress);
+      if (!vec.empty()) {
+        const auto subnet = folly::IPAddress::tryFromString(vec.at(0)).value();
+        return Status::UserError(
+            "CIDR value '{}' is > network bit count '{}'",
+            vec.size() == 2 ? vec.at(1)
+                            : folly::to<std::string>(
+                                  subnet.isV4() ? ipaddress::kIPV4Bits
+                                                : ipaddress::kIPV6Bits),
+            subnet.bitCount());
+      }
+      return Status::UserError(
+          "Invalid IP address of size:{} received", ipaddress.size());
+    }
+    default:
+      return Status::UserError(
+          "Unknown parsing error when parsing IP address: {} ", ipaddress);
+  }
+}
+} // namespace
+
+inline folly::Expected<std::pair<int128_t, int8_t>, Status>
+tryParseIpPrefixString(const std::string& ipprefixString) {
+  // Ensure '/' is present
+  if (ipprefixString.find('/') == std::string::npos) {
+    return folly::makeUnexpected(
+        threadSkipErrorDetails()
+            ? Status::UserError()
+            : Status::UserError(
+                  "Invalid CIDR IP address specified. Expected IP/PREFIX format, got: {}",
+                  ipprefixString));
+  }
+
+  auto tryCdirNetwork = folly::IPAddress::tryCreateNetwork(
+      ipprefixString, /*defaultCidr*/ -1, /*applyMask*/ false);
+  if (tryCdirNetwork.hasError()) {
+    return folly::makeUnexpected(handleFailedToCreateNetworkError(
+        ipprefixString, std::move(tryCdirNetwork.error())));
+  }
+
+  folly::ByteArray16 addrBytes;
+  auto cdirNetwork = std::move(tryCdirNetwork.value());
```

Comment:
Should we 
`const auto& cdirNetwork = tryCdirNetwork.value();`
instead?

Is someone to access tryCdirNetwork after this line - would be problem and there is no need for that move at all or new local variable, except for ease of access (cdirNetwork VS tryCdirNetwork.value()).